### PR TITLE
change dcr resource naming convention used in aks monitoring addon

### PR DESCRIPTION
**Pull request by ganga1980**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by daweim0 — APPROVED** [review]

File: src/aks-preview/HISTORY.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.5.42
+++++++
+* In AKS Monitoring addon, fix DCR resource naming convention from DCR-<workspaceName> to MSCI-<workspaceName> to make consistent naming across.
+
 0.5.42
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.md:9]: We already have 0.5.42 and a 0.5.43 ([PR #4054](https://github.com/Azure/azure-cli-extensions/pull/4054)) pending merge. Consider 0.5.44?

Besides, please also change the version [here](https://github.com/ganga1980/azure-cli-extensions/blob/main/src/aks-preview/setup.py#L11).

Comment (ganga1980) [reply src/aks-preview/HISTORY.md]: > We already have 0.5.42 and a 0.5.43 ([PR #4054](https://github.com/Azure/azure-cli-extensions/pull/4054)) pending merge. Consider 0.5.44?
> 
> Besides, please also change the version [here](https://github.com/ganga1980/azure-cli-extensions/blob/main/src/aks-preview/setup.py#L11).

> @FumingZhang , fixed the feedback.

**Review by wangzelin007 — APPROVED** [review]

**Comment by ganga1980** [comment]
Hi, @wangzelin007 , can you please help on merging this PR?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**